docker/conf/homeserver.yaml

```diff
@@ -57,8 +62,11 @@ database:
     user: "{{ POSTGRES_USER or "synapse" }}"
     password: "{{ POSTGRES_PASSWORD }}"
     database: "{{ POSTGRES_DB or "synapse" }}"
+{% if not SYNAPSE_USE_UNIX_SOCKET %}
+{# Fun fact: if left with no host or port, Synapse looks for the default Unix socket instead. #}
```

Comment:
What's the default unix socket? (sorry, I'm not familiar)

Default unix socket location for Postgres? Is this a behavior of `psycopg2`?

Reply:
(It's quite alright) The default configuration for a Unix socket for Postgresql puts it at `/var/run/postgresql/.s.PGSQL.5432`

And yes, it appears to be from `psycopg2`(full disclosure, I didn't investigate to deeply into exactly how, but I did find [this](https://github.com/psycopg/psycopg2/blob/9f020124f83ed04d0d213eca7735555f618fcd47/lib/__init__.py#L98) in their source)

Reply:
```suggestion
{# Synapse will use a default unix socket for Postgres when host/port is not specified (behavior from `psycopg2`). #}
```